src/Microsoft.Framework.FileSystemGlobbing/Internal/PathSegments/LiteralPathSegment.cs

```diff
@@ -29,7 +33,7 @@ public override bool Equals(object obj)
             var other = obj as LiteralPathSegment;
 
             return other != null &&
-                string.Equals(other.Value, Value, StringComparison.Ordinal);
+                string.Equals(other.Value, Value, _comparisonType);
```

Comment:
Would it suffice to simply `Debug.Assert` in that case? It would be an odd case for the two comparisons to be out of sync, wouldn't it?